lambda_strip_docx/lambda_function.py

```diff
@@ -84,6 +84,13 @@ def lambda_handler(event, context):
                     logger.warning(f"Skipping non-DOCX file: {object_key}")
                     continue
 
+                # Check if the file has already been processed
+                tags_response = s3.get_object_tagging(Bucket=bucket_name, Key=object_key)
+                tags = {tag['Key']: tag['Value'] for tag in tags_response.get('TagSet', [])}
+                if 'DOCUMENT_PROCESSOR_VERSION' in tags and tags['DOCUMENT_PROCESSOR_VERSION'] == document_processor_version:
```

Comment:
I think we should be clear what we mean by the DOCUMENT_PROCESSOR_VERSION -- an upgrade from 1.0.0 to 1.0.1 or similar, or downgrade, or `-dev` suffix will trigger reprocessing if requested. 

This is probably OK because we'll probably only bulk process if there's a significant change in the cleaning process, but I think this might not want to be a semantic number.

Reply:
We should try to standardise this behaviour between the various things (parse/enrich/cleanse) with potential for re-running. At the moment I we just use major and minor versions (skipping patch entirely), where a major version bump should prompt re-running the entire corpus but a minor version bump is just "we made it a bit better or fixed a bug but don't worry about reprocessing".

Reply:
we agreed offline that semver works well and only to skip processing if major version matches